_source/_posts/2023-09-14-spring-boot-keycloak-kubernetes.md

```diff
@@ -0,0 +1,725 @@
+---
+layout: blog_post
+title: "Identity in Spring Boot with Kubernetes, Keycloak, and Auth0"
+author: jimena-garbarino
+by: contractor
+communities: [devops,security,java]
+description: "A walk-through of building a microservices architecture with JHipster, Keycloak, Auth0, and Google Kubernetes Engine (GKE) deployment."
+tags: [keycloak, kubernetes, java, jhipster, security, oauth2]
+tweets:
+- ""
+- ""
+- ""
+image:
+type: awareness
+---
+JHipster is a powerful development platform that provides developers with the tools they need to create modern, scalable, and robust web applications using Spring Boot. Using OAuth 2.0 with JHipster is an excellent way to ensure your web applications are secure and well-protected from unauthorized access. Integrating OAuth and OpenID Connect (OIDC) into your applications allows you to easily implement authorization and authentication, manage user identities, and provide secure access to resources.
+
+By default, the OAuth 2.0 and OIDC server configured with JHipster is Keycloak, an open source identity and access management solution. This is because Keycloak can be run in a local Docker container and pre-populated with users and client registrations. And while you can also [use JHipster with Auth0 directly](https://www.jhipster.tech/security/#auth0), and we've written a [couple](https://auth0.com/blog/full-stack-java-with-react-spring-boot-and-jhipster/) of [tutorials](https://auth0.com/blog/micro-frontends-for-java-microservices/) to help you migrate a JHipster project to Auth0.
```

Comment:
```suggestion
By default, the OAuth 2.0 and OIDC server configured with JHipster is Keycloak, an open source identity and access management solution. This is because Keycloak can be run in a local Docker container and pre-populated with users and client registrations. You can also [use JHipster with Auth0 directly](https://www.jhipster.tech/security/#auth0) and we've written a [couple](https://auth0.com/blog/full-stack-java-with-react-spring-boot-and-jhipster/) of [tutorials](https://auth0.com/blog/micro-frontends-for-java-microservices/) to help you migrate a JHipster project to Auth0.
```